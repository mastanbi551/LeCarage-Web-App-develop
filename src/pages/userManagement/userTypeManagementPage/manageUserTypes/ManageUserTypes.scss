@import "../../../../styles//variables.scss";

.manage-usertypes-page {
  text-align: left;
  .add-button {
    margin-bottom: 125px;
  }
  .row {
    padding: 10px 5px;
  }
  .details-container {
   // padding: 15px 20px;

    &__details-container-header {
      svg {
        right: 10px !important;
      }
    }
  }

  // .heading {
  //     //height: 0px;
  // }

  &__fetaures-info {
    padding-top: 30px;

    .form-checkbox {
      border: 1px solid $primaryGreen;
      border-radius: 3px;
    }
  }

  .input-text-field {
    .pen-icon {
      position: absolute;
      right: 10px;
      top: 10px;
      color: $black;
    }
  }

  input {
    width: 100%;
    padding: 10px;
  }

  .label {
    @include label;
    color: $black;
  }

  .feature-heading {
    @include montserrat16;
    letter-spacing: 0.15px;
    color: $primaryBlack;
    margin: 0px;
    margin-bottom: 5px;
  }

  .feature-check-boxes {
    margin-bottom: 5px;
  }

  .feature-info {
    padding-bottom: 0px;
  }

  .checkbox-btn {
    color: $black;
    display: inline-block;
    padding-right: 20px;

    .checkboxlist,
    .checkbox-text {
      display: inline-block;
    }

    .checkbox-text {
      @include input-text;
      line-height: 20px;
      color: $primaryBlack;
      letter-spacing: 0.25px;
    }
  }

  .heading {
    @include SubHeading2;
  }

  .add-button {
    margin-bottom: 125px;
  }

  .feature-radio input {
    width: 20px;
    display: inline-block;
    accent-color: $primaryGreen;

    :hover {
      accent-color: $primaryGreen;
    }
  }
}

.dropdown {
  margin-bottom: 1px !important;
}
