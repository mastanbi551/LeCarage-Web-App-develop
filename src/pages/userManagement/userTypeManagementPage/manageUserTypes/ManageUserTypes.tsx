import React, { useState } from 'react';
import {
  InputTextField,
  PageContainer,
  DetailsContainer,
  CardContainer,
  Loader,
} from '../../../../components';
import './ManageUserTypes.scss';
import { ConstantStrings } from '../../../../utils/constants';
import { Formik } from 'formik';
import { UserTypesPageValidation } from '../../../../utils/ValidationSchemas';
import { CustomRadioButtonList } from '../../../../components/index';
import _ from 'lodash';
//ManageUserTypes page handles create Usertypes functionality and update Usertypes functionality
interface ManageUserTypesInterfaceProps {
  backButtonHandler: () => void;
  userTypeData: {
    userTypeId: string;
    userType: string;
    description: string;
    statusId: string;
    features: {
      featureId: string;
      accessId: string;
    }[];
  };
  statusDropDownData: { label: string; value: string }[];
  isDisable: boolean;
  featuresList: { label: string; value: string }[];
  featureAccessList: { label: string; value: string }[];
  isEditable: boolean;
  isCreate: boolean;
  handleRadioButtonClick: (e: React.ChangeEvent<HTMLInputElement>,
    featureValue: string,
    feature:  {
      value: string,
      label: string
    },
    access: {
      value: string,
      label: string
    }) => void;
  saveOrUpdateButtonHandler: (values: {
    userTypeId: string,
    userType: string,
    description: string
    statusId: string,
    features: {
      featureId: string,
      accessId: string
    }[],
  }) => void;
  selectedFeatureAccessList: {
    accessId: string,
    featureId: string
  }[],
  handleReset: (values: {
    userTypeId: string,
    userType: string,
    description: string
    statusId: string,
    features: {
      featureId: string,
      accessId: string
    }[],
  }) => void;
}
export const ManageUserTypes: React.FC<ManageUserTypesInterfaceProps> = ({
  backButtonHandler,
  userTypeData,
  statusDropDownData,
  isDisable,
  featuresList,
  featureAccessList,
  isEditable,
  isCreate,
  handleRadioButtonClick,
  saveOrUpdateButtonHandler,
  selectedFeatureAccessList,
  handleReset
}) => {
  const [submitClick, setSubmitClick] = useState(false);

  return (
    <>
      {((isEditable || isDisable) && userTypeData.userTypeId !== '') || isCreate ? (
        <PageContainer
          mainHeadingName='User Types'
          backIcon={true}
          backIconOnClick={backButtonHandler}
        >
          <Formik
            initialValues={{
              userTypeId: _.get(userTypeData, 'userTypeId') ? _.get(userTypeData, 'userTypeId') : '',
              userType: _.get(userTypeData, 'userType') ? _.get(userTypeData, 'userType') : '',
              description: _.get(userTypeData, 'description') ? _.get(userTypeData, 'description') : '',
              statusId: _.get(userTypeData, 'statusId') ? _.get(userTypeData, 'statusId') : '',
              features: _.get(userTypeData, 'features') ? _.get(userTypeData, 'features') : []
            }}
            validationSchema={UserTypesPageValidation}
            onSubmit={(values) => {
              saveOrUpdateButtonHandler(values);
            }} 
            onReset={(values) => {
              values.features =  _.get(userTypeData, 'features') ? _.get(userTypeData, 'features') : [];
              handleReset(values);
            }}           
             >            
            {({
              handleChange,
              handleSubmit,
              values,
              errors,
              resetForm,
              isValid,
              handleBlur,
              touched,
            }) => (
              <>
                <form className='container-fluid' onSubmit={handleSubmit}>
                  <div className='row manage-usertypes-page'>
                    <div className='col-md-8'>
                      <div className='col-md-12'>
                        <DetailsContainer
                          heading='User Type Details'
                          isCollapsible={false}
                        >
                          <>
                            <div className='manage-usertypes-page__usertype-info'>
                              <div className='row'>
                                <div className='col-md-6'>
                                  <label className='label'>
                                    {
                                      ConstantStrings.manageUserType
                                        .userTypeName
                                    }
                                  </label>
                                  <InputTextField
                                    name='user-type-name-inputField'
                                    fieldName='usertypename'
                                    id=''
                                    placeholder=''
                                    inputType='text'
                                    requiredOrNot={true}
                                    value={values && values.userType}
                                    onChangeText={handleChange('userType')}
                                    isDisable={isDisable}
                                    onBlur={handleBlur('userType')}
                                    error={errors.userType}
                                    touched={touched.userType}
                                  />
                                </div>
                              </div>
                              <div className='row'>
                                <div className='col-md-12'>
                                  <label className='label'>
                                    {ConstantStrings.manageUserType.description}
                                  </label>
                                  <InputTextField
                                    name='user-type-description-inputField'
                                    fieldName='usertypeDescription'
                                    id=''
                                    placeholder=''
                                    inputType='text'
                                    requiredOrNot={true}
                                    value={values.description}
                                    onChangeText={handleChange('description')}
                                    isDisable={isDisable}
                                    onBlur={handleBlur('description')}
                                    error={errors.description}
                                    touched={touched.description}
                                  />
                                </div>
                              </div>
                            </div>

                            <div className='manage-usertypes-page__fetaures-info'>
                              <h1 className='heading'>
                                {ConstantStrings.manageUserType.userTypeDetails}
                              </h1>

                              {featuresList &&
                                featuresList.map((feature: { label: string, value: string}, index: number) => {
                                  const result = _.filter(selectedFeatureAccessList, {
                                    featureId: feature.value,
                                  });

                                  return (
                                    <div
                                      key={feature.value}
                                      className='feature-heading'
                                    >
                                      <h1 className='feature-heading'>
                                        {feature.label}
                                      </h1>
                                      <div className='row'>
                                        
                                        {
                                          isEditable || isDisable ?
                                            <CustomRadioButtonList
                                              radioButtonList={featureAccessList}
                                              name=''
                                              handleRadioButtonLClick={() => {
                                                setSubmitClick(false);
                                                handleRadioButtonClick;
                                              }}
                                              id={feature.value}
                                              defaultChecked={result.length > 0 ? result[0].accessId : -1}
                                              values={selectedFeatureAccessList}
                                              feature={feature}
                                              isDisable={isDisable}
                                          />
                                          :
                                          <CustomRadioButtonList
                                          radioButtonList={featureAccessList}
                                          name=''
                                          handleRadioButtonLClick={handleRadioButtonClick}
                                          id={feature.value}
                                          //defaultChecked={result.length > 0 ? result[0].accessId : -1}
                                          values={selectedFeatureAccessList}
                                          feature={feature}
                                          isDisable={isDisable}
                                      />

                                        }
                                        
                                        
                                      </div>
                                    </div>
                                  );
                                })}

                              {selectedFeatureAccessList.length === 0 && submitClick &&
                                <div className='validation-error-message text-align-left'>
                                {'select usertype details'}
                              </div>
                              }
                                
                            </div>
                          </>
                        </DetailsContainer>
                      </div>
                    </div>
                    <div className='col-md-4 details-card-containers'>
                      <div className='col-md-12'>
                        <CardContainer
                          heading='User Type Actions'
                          showStatusSelectText={true}
                          statusDropDownList={statusDropDownData}
                          statusDropDownOnChange={handleChange('statusId')}
                          errorStatus={errors.statusId}
                          touchedStatus={touched.statusId}
                          onBlurStatus={handleBlur('statusId')}
                          statusDefaultValue={isEditable ? values.statusId : values.statusId !== '' ? values.statusId : ''}
                          onSaveAndUpdateButtonOnClick={() => {
                            setSubmitClick(true);
                            handleSubmit();
                          }}
                          onDiscardButtonOnClick={() => {
                            resetForm();
                            handleReset(values);
                          }}
                          isValid={isValid}
                          isDisable={isDisable}
                        />
                      </div>
                    </div>
                  </div>
                </form>
              </>
            )}
          </Formik>
        </PageContainer>
      ) : (
        <Loader />
      )}
    </>
  );
};
