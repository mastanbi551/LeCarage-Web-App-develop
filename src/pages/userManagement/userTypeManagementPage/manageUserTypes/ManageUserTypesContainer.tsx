import React, { useEffect, useState } from 'react';
import { useLocation, useNavigate } from 'react-router-dom';
import { ManageUserTypes } from './ManageUserTypes';
import _, { merge } from 'lodash';
import { useAppDispatch, useAppSelector } from '../../../../hooks/Hooks';

import {
  createUserTypeRequest,
  updateUserTypeRequest,
  getFeatureAccessList,
  getFeatures,
  viewUserTypeRequest,
  reset,
} from '../userTypeManagementPageState/UserTypeManagementSlice';

import { toast } from 'react-toastify';
import { AccessTypes, responseStatus, routeNames } from '../../../../utils/constants/ConstantStrings';
import { NotFound } from '../../../notFound/NotFound';
import { getStatusDetails } from '../../../commonStates/CommonSlice';
import { Loader } from '../../../../components';

export interface FeatureListProps {
  label: string;
  value: string;
  parentFeature: string;
  parentFeatureId: string;
}
export const ManageUserTypesContainer = () => {
  const location = useLocation();
  const dispatch = useAppDispatch();
  const navigate = useNavigate();

  const [userInfo, setUserInfo] = useState(
    location.state ? location.state.userInfo : ''
  );

  const [isCreate, setIsCreate] = useState(
    location.state && location.state.action === 'create' ? true : false
  );

  const [isEditable, setIsEditable] = useState(
    location.state &&
      location.state.userInfo &&
      location.state.action === 'edit'
      ? true
      : false
  );

  const [isDisable, setIsDisable] = useState(
    location.state &&
      location.state.userInfo &&
      (location.state.action === 'view')
      ? true
      : false
  );

  const [featuresInfo, setFeaturesInfo] = useState<
    { label: string; value: string }[]
  >([]);
  const [accessInfo, setAccessInfo] = useState<
    { label: string; value: string }[]
  >([]);
  const [statusDropdownData, setStatusDropdownData] = useState<
    { label: string; value: string }[]
  >([]);

  const [selectedFeatureAccessList, setSelectedFeatureAccessList] = useState<
    { featureId: string; accessId: string }[]
  >([]);

  const {
    featureList,
    featureAccessList,
  
    isLoading,
    responseData,

    viewUserTypeIsLoading,
    viewUserTyperesponseData,

    deleteUserTypeIsLoading,
    deleteUsertypeResponseData,
  } = useAppSelector((state) => state.userTypeManagementReducer);
  
  const {statusList} = useAppSelector(state => state.commonReducer);
  const [userTypeData, setuserTypeData] = useState({
    userTypeId: '',
    userType: '',
    description: '',
    statusId: '',
    features: [{ featureId: '', accessId: ''}],
  });

  const {routes} = useAppSelector(state => state.sidebarReducer);
  const [ access, setAccess] = useState('');
  
  useEffect(() => {
    if(access === '' && routes){
      const SubRoutes = routes ? _.get(routes, routeNames.userManagement) : 0;
      if(SubRoutes){
        const getAccess = _.filter(SubRoutes, (obj: {
          accessId: string;
          featureAccess: string;
          isParent: boolean;
          label: string;
          parentFeature: string;
          value: string;
        }) => obj.label === routeNames.usertypes);
        if(getAccess.length > 0){
          setAccess(getAccess ? getAccess[0].featureAccess : '');
        }
      }
      
    }
  }, [access, routes]);

  useEffect(() => {
    if (
      userTypeData.userTypeId === '' &&
      userInfo !== '' &&
      (isEditable || isDisable)
    ) {
      dispatch(reset());
      const params = { userTypeId: _.get(userInfo, 'User Type ID') };
      dispatch(viewUserTypeRequest(params));
      setUserInfo('');
    }
  }, [userTypeData, userInfo]);

  useEffect(() => {
    if (viewUserTypeIsLoading) {
      if (viewUserTyperesponseData.status === responseStatus.success) {
        setuserTypeData(viewUserTyperesponseData.details[0]);
        setSelectedFeatureAccessList(viewUserTyperesponseData.details[0]?.features);
      }
    }
    if (deleteUserTypeIsLoading) {
      if (deleteUsertypeResponseData.status === responseStatus.success) {
        navigate('/usertypes');
      }
    }
  }, [
    viewUserTypeIsLoading,
    viewUserTyperesponseData,
    deleteUserTypeIsLoading,
    deleteUsertypeResponseData,
  ]);

  useEffect(() => {
    if (isLoading) {
      if (responseData.status === responseStatus.success) {
        dispatch(reset());
        navigate('/usertypes');
      } else if (responseData.status === responseStatus.fail) {
        toast.error(responseData.errDescription, {
          position: toast.POSITION.TOP_CENTER,
          className: 'toast-message',
        });
      }
    }
  }, [isLoading, responseData]);

  useEffect(() => {
    if(featureList.length === 0){
      dispatch(getFeatures());
    }
    else {
      setFeaturesInfo(featureList);
    }
  }, [featureList]);

  useEffect(() => {
    if(featureAccessList.length === 0){
      dispatch(getFeatureAccessList());
    }
    else {
      setAccessInfo(featureAccessList);
    }
  }, [featureAccessList]);

  useEffect(() => {   
    if (statusList.length === 0) {
      dispatch(getStatusDetails());
    }
    else {
      setStatusDropdownData(statusList);
    }
  }, [statusList]);
  
  function handleFeatureUpdate(
    featureId: string,
    accessId: string,
    feature: {
      value: string,
      label: string
    },
    access: {
      value: string,
      label: string
    }
  ) {
    const length = _.filter(
      userTypeData.features,
      (obj: {featureId: string, accessId: string}) => obj.featureId === feature.value
    ).length;

    if (length > 0) {
      const updatedObject = {
        ...userTypeData,
        features: userTypeData.features.map((feature: {featureId: string, accessId: string}) => {
          if (feature.featureId === featureId) {
            return {
              ...feature,
              accessId: accessId,
            };
          } else {
            return feature;
          }
        }),
      };
      setuserTypeData(updatedObject); 
    } else {
      _.remove(
        userTypeData.features,
        (obj: {featureId: string, accessId: string}) => obj.featureId === '' || obj.featureId === feature.value
      );
      const newFeature = { featureId: feature.value, accessId: access.value };

      setuserTypeData(prevState => ({
        ...prevState,
        features: [...prevState.features, newFeature]
      }));
    }
  }

  const handleRadioButtonClick = (
    e: React.ChangeEvent<HTMLInputElement>,
    featureValue: string,
    feature:  {
      value: string,
      label: string
    },
    access: {
      value: string,
      label: string
    }
  ) => {
    if (e.target.checked) {
      if (isEditable) {
        handleFeatureUpdate(
          featureValue,
          e.target.value,
          feature,
          access
        );
      } else {
        _.remove(
          selectedFeatureAccessList,
          (obj: {featureId: string, accessId: string}) => obj.featureId === '' || obj.featureId === featureValue
        );
        const newObject = {
          featureId: featureValue,
          accessId: access.value,
        };
        selectedFeatureAccessList.push(newObject);
        setSelectedFeatureAccessList(selectedFeatureAccessList);
      }
      // setCheckedList(values.features);
    } else {
      _.remove(selectedFeatureAccessList, (obj: {featureId: string, accessId: string}) => obj.featureId === featureValue);
    }
  };

  const backButtonHandler = () => {
    history.back();
  };

  const saveOrUpdateButtonHandler = (values: {
    userTypeId?: string | undefined;
    userType: string;
    description: string;
    statusId: string;
    features: {
      featureId: string;
      accessId: string;
    }[];
  }) => {
    if (isEditable) {
      const params = JSON.stringify(merge({}, values, { features: userTypeData.features }));
      const formatJson = params.replace(/'/g, '\'\'');
      dispatch(updateUserTypeRequest(formatJson));
    } else {
      // /_.filter(data, (obj) => obj.parentFeature !== "Dashboard");
      const distinctParentFeaturesNew =  _.filter(_.uniqBy(featureList, 'parentFeatureId'), (obj: FeatureListProps) => obj.parentFeature !== 'Dashboard');
      const distinctParentFeatures = distinctParentFeaturesNew.map((item: FeatureListProps) => {
          return {
            featureId: item.parentFeatureId,
            accessId: '1'
          };       
      });
      const updatedObject = merge({}, values, { features: _.unionBy(selectedFeatureAccessList, distinctParentFeatures) });
      const params = JSON.stringify(_.omit(updatedObject, 'userTypeId'));
      const formatJson = params.replace(/'/g, '\'\'');
      dispatch(createUserTypeRequest(formatJson));
    }
  };


  const handleReset= () => {  
    isCreate &&  
      setSelectedFeatureAccessList([]);
    isEditable &&
      setSelectedFeatureAccessList(viewUserTyperesponseData?.details[0]?.features);
  };

  if (access !== '') {
  if(access === AccessTypes.fullAccess || access === AccessTypes.readOnlyAccess){
    return (
      <ManageUserTypes
        backButtonHandler={backButtonHandler}
        userTypeData={userTypeData}
        statusDropDownData={statusDropdownData}
        isDisable={isDisable}
        featuresList={featuresInfo}
        featureAccessList={accessInfo}
        isEditable={isEditable}
        isCreate={isCreate}
        handleRadioButtonClick={handleRadioButtonClick}
        saveOrUpdateButtonHandler={saveOrUpdateButtonHandler}
  
        selectedFeatureAccessList={selectedFeatureAccessList}
        handleReset={handleReset}
      />
    );
  }
  else {
    return (
      <NotFound />
    );
  }
} else {
  return <Loader />;
}
};
