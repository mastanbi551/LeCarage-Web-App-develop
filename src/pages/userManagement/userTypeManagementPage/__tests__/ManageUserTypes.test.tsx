import React from 'react';
import renderer from 'react-test-renderer';
import { ManageUserTypes } from '../manageUserTypes/ManageUserTypes';

describe('ManageUserTypes component', () => {
  const mockBackButtonHandler = jest.fn();
  const mockSaveOrUpdateButtonHandler = jest.fn();
  const mockHandleReset = jest.fn();

  const props = {
    backButtonHandler: mockBackButtonHandler,
    userTypeData: {
      userTypeId: '1',
      userType: 'Admin',
      description: 'Administrator',
      statusId: '1',
      features: [
        {
          featureId: '1',
          accessId: '1',
        },
      ],
    },
    statusDropDownData: [
      {
        label: 'Active',
        value: '1',
      },
    ],
    isDisable: true,
    featuresList: [
      {
        label: 'Feature 1',
        value: '1',
      },
    ],
    featureAccessList: [
      {
        label: 'Access 1',
        value: '1',
      },
    ],
    isEditable: true,
    isCreate: false,
    handleRadioButtonClick: jest.fn(),
    saveOrUpdateButtonHandler: mockSaveOrUpdateButtonHandler,
    selectedFeatureAccessList: [
      {
        accessId: '1',
        featureId: '1',
      },
    ],
    handleReset: mockHandleReset,
  };

  it('renders correctly', () => {
    const tree = renderer.create(<ManageUserTypes {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
