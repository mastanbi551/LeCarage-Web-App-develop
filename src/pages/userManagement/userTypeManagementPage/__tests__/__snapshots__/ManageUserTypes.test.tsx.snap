// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ManageUserTypes component renders correctly 1`] = `
<div
  className="page-container-details"
>
  <div
    className="page-container-details__header-content"
  >
    <svg
      className="back-icon"
      data-testid="back-icon"
      fill="none"
      height="1em"
      onClick={[MockFunction]}
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <line
        x1="19"
        x2="5"
        y1="12"
        y2="12"
      />
      <polyline
        points="12 19 5 12 12 5"
      />
    </svg>
    <h1
      className="page-container-details__header-content__heading"
    >
      User Types
    </h1>
  </div>
  <form
    className="container-fluid"
    onSubmit={[Function]}
  >
    <div
      className="row manage-usertypes-page"
    >
      <div
        className="col-md-8"
      >
        <div
          className="col-md-12"
        >
          <div
            className="details-container"
          >
            <div
              className="details-container__details-container-header"
            >
              <h1
                className="details-container__details-container-header__heading"
              >
                User Type Details
              </h1>
            </div>
            <div
              className="manage-usertypes-page__usertype-info"
            >
              <div
                className="row"
              >
                <div
                  className="col-md-6"
                >
                  <label
                    className="label"
                  >
                    User Type Name
                  </label>
                  <div
                    className="input-text-field"
                  >
                    <input
                      autoComplete="new-password"
                      autoCorrect="off"
                      className="user-type-name-inputField static"
                      disabled={true}
                      id=""
                      name="usertypename"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      placeholder=""
                      required={true}
                      type="text"
                      value="Admin"
                    />
                  </div>
                </div>
              </div>
              <div
                className="row"
              >
                <div
                  className="col-md-12"
                >
                  <label
                    className="label"
                  >
                    Description
                  </label>
                  <div
                    className="input-text-field"
                  >
                    <input
                      autoComplete="new-password"
                      autoCorrect="off"
                      className="user-type-description-inputField static"
                      disabled={true}
                      id=""
                      name="usertypeDescription"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      placeholder=""
                      required={true}
                      type="text"
                      value="Administrator"
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              className="manage-usertypes-page__fetaures-info"
            >
              <h1
                className="heading"
              >
                User Type Details
              </h1>
              <div
                className="feature-heading"
              >
                <h1
                  className="feature-heading"
                >
                  Feature 1
                </h1>
                <div
                  className="row"
                >
                  <div
                    className="radiobutton row"
                  >
                    <div
                      className="col-md-3 feature-radio"
                    >
                      <input
                        checked={true}
                        className=""
                        disabled={true}
                        id="feature_1_access_1"
                        name="feature_1_access"
                        onChange={[Function]}
                        type="radio"
                        value="1"
                      />
                      <label
                        htmlFor="feature_1_access_1"
                      >
                        Access 1
                      </label>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="col-md-4 details-card-containers"
      >
        <div
          className="col-md-12"
        >
          <div
            className="card-container"
          >
            <div
              className="col-md-12"
            >
              <div
                className="card-container__image-wrapper"
              />
               
            </div>
            <h1
              className="card-container__heading"
            >
              User Type Actions
            </h1>
            <div
              className="dropdown input-text-field disabled"
            >
              <label
                className="dropDown-label"
              >
                Set Status
              </label>
              <select
                className="custom-select status-dropdown"
                defaultValue="1"
                disabled={true}
                onBlur={[Function]}
                onChange={[Function]}
                value="1"
              >
                <option
                  className="custom-option"
                  value=""
                >
                  Select
                </option>
                <option
                  className="custom-option"
                  selected={true}
                  value="1"
                >
                  <span
                    className="inactive-icon"
                  />
                  <span>
                    Active
                  </span>
                </option>
              </select>
            </div>
            <div
              className="line"
            />
            <button
              className="greenButton w-100"
              disabled={true}
              onClick={[Function]}
              type="button"
            >
              Save & Update
            </button>
            <button
              className="add-button w-100"
              disabled={true}
              onClick={[Function]}
              type="button"
            >
              Discard
            </button>
          </div>
        </div>
      </div>
    </div>
  </form>
</div>
`;
